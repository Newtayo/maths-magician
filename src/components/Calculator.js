import React from 'react';
import ButtonBox from './ButtonBox';
import Screen from './Screen';

const Calculator = () => (
  <div>
    <Screen />
    <ButtonBox />
  </div>
);
export default Calculator;
