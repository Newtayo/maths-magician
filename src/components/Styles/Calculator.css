.calcudisplay {
  display: flex;
  width: 100%;
}

.calculatorheader {
  width: 50%;
  text-align: left;
  margin-right: 5%;
}
