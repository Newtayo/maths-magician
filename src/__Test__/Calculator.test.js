import React from 'react';
import '@testing-library/jest-dom';
import { render, fireEvent } from '@testing-library/react';

import Calculator from '../components/Calculator';

test('renders correctly', () => {
  const { asFragment } = render(<Calculator />);
  expect(asFragment()).toMatchSnapshot();
});

//

test('should display the number click on the calculator button to the screen', async () => {
  const { getByTestId, getByText } = render(<Calculator />);
  fireEvent.click(getByText('4'));
  const lot = getByTestId('screen');

  expect(lot.textContent).toBe('4');
});
