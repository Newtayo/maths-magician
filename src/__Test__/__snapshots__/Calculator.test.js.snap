// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="calcudisplay"
  >
    <h2
      class="calculatorheader"
    >
      Let' s do some math!
    </h2>
    <div
      class="display"
    >
      <div
        class="screen"
        data-testid="screen"
      >
        0
      </div>
      <button
        class="undefined btn"
        type="submit"
      >
        AC
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        +/-
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        %
      </button>
      <button
        class="opt btn"
        type="submit"
      >
        รท
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        7
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        8
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        9
      </button>
      <button
        class="opt btn"
        type="submit"
      >
        x
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        4
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        5
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        6
      </button>
      <button
        class="opt btn"
        type="submit"
      >
        -
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        1
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        2
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        3
      </button>
      <button
        class="opt btn"
        type="submit"
      >
        +
      </button>
      <button
        class="zero btn"
        type="submit"
      >
        0
      </button>
      <button
        class="undefined btn"
        type="submit"
      >
        .
      </button>
      <button
        class="opt btn"
        type="submit"
      >
        =
      </button>
    </div>
  </div>
</DocumentFragment>
`;
